lab-derek/test/test-index.js

```diff
@@ -0,0 +1,184 @@
+'use strict';
+
+const superagent = require('superagent');
+const expect = require('expect');
+
+const server = require('../lib/server.js');
+
+//TODO: your tests should start your server when they begin and stop your server when they finish
+
+describe('testing task routes', () => {
+  before((done) => {
+    server.listen(3000, () => {
+      console.log('server up for testing');
+      done();
+    });
+  });
+  after((done) => {
+    server.close(() => {
+      console.log('testing finished, server closed');
+      done();
+    });
+  });
+  //TODO: write a test to ensure that your api returns a status code of 404 for routes that have not been registered
+  it('should return a status code 404 for unregistered routes', () => {
```

Comment:
Need to pass done into this callback. And all other 'it' callbacks. Otherwise you will get false positives